refactor: Use `Typography` from `@equinor/mad-components`

dansvend opened the pull request:
- Replace `Text` from `react-native`
- Replace `Typography` from `mad-expo-core`
- Use Equinor font for navigation header title and header back title
- Delete project implementation of `Typography`
- Consistent capitalization of `README.md` files
- Update `README.md` files for component types

Closes #296

Sandtory commented on components/common/atoms/Typography.tsx:
```diff
@@ -1,59 +0,0 @@
-
-/**
- * The new design for this app only has 3 different textvariants: Big headers, small headers, and paragraphs.
- * I have therefore only made 3 variants
```
Comments like these are good in a code review scenario but it might not be necessary to justify things already written in the code 🫡

Sandtory commented on screens/SoundCheckFinishedScreen.tsx:
```diff
@@ -1,10 +1,9 @@
-import Typography from "../components/common/atoms/Typography";
 import { RootStackScreenProps } from "../types";
 
 const styles = StyleSheet.create({
```
I think its best to keep the stylesheets at the bottom to increase readability

dansvend replied:
I totally agree, and well spotted 👌 To avoid scope creep I'm sticking to the issue description, but created a separate [issue](https://github.com/equinor/hearing-test/issues/307) for this 💪

Sandtory commented on screens/SoundCheckScreen.tsx:
```diff
@@ -7,7 +7,6 @@ import Sound from "react-native-sound";
 import SystemSetting from "react-native-system-setting";
 
```
Im only allowed to comment on changes, but the same changes as before are needed here, stylesheets should be moved to the bottom of the file 🤩

dansvend replied:
Nicely spotted 🔍 This will be covered by #307 📦

Sandtory commented on screens/SoundCheckScreen.tsx:
```diff
@@ -7,7 +7,6 @@ import Sound from "react-native-sound";
 import BigRoundButton from "../components/common/atoms/BigRoundButton";
-import Typography from "../components/common/atoms/Typography";
 import ProgressAnimationBar from "../components/common/molecules/ProgressAnimationBar";
 import { EQUINOR_GREEN, GRAY_BACKGROUND } from "../constants/colors";
```
Also some of the components can be exported upon initialization instead of at the end of the file, you should look into this

dansvend replied:
I agree - named exports makes it easier to refactor the codebase 👌 Not a part of the scope of this issue so I made a separate [issue](https://github.com/equinor/hearing-test/issues/308) for this 💪

Sandtory commented on screens/TestLogScreen.tsx:
```diff
@@ -1,6 +1,7 @@
+import { Typography } from "@equinor/mad-components";
 import { Spinner } from "mad-expo-core";
```
Look into the deprecation of this spinner and consider using the one from mad components

dansvend replied:
Good suggestion 🙌 I've already changed this for this case in [this](https://github.com/equinor/hearing-test/pull/305) Pull Request, but I opened a new [issue](https://github.com/equinor/hearing-test/issues/309) for the rest of the codebase 👌

Sandtory commented on screens/TestScreen.tsx:
```diff
@@ -1,4 +1,4 @@
-import { Button } from "@equinor/mad-components";
+import { Button, Typography } from "@equinor/mad-components";
```
Many TS errors here, should be looked into, but unrelated to this PR

dansvend replied:
Thanks for highlighting this 🔦 I created an [issue](https://github.com/equinor/hearing-test/issues/310) to fix this 💪

Sandtory reviewed (commented):
The code here looks good, the move from the old typography to the new one is a good choice. See the comments i added and i will approve the pull request 🤠🫡

Sandtory reviewed (commented):
The code here looks good, the move from the old typography to the new one is a good choice. See the comments i added and i will approve the pull request 🤠🫡

dansvend commented:
> The code here looks good, the move from the old typography to the new one is a good choice. See the comments i added and i will approve the pull request 🤠🫡

Thank you brother 🤝

I agree, the code will be more maintainable, easier to review and look cleaner after migrating to components from [@equinor/mad-components](https://www.npmjs.com/package/@equinor/mad-components) 👌

Thanks for your review and I like your eye for details 🔍  I added separate issues for your suggestions and will fix them in new Pull Requests 💪
